adam-core/src/main/scala/org/bdgenomics/adam/hbase/HBaseFunctions.scala

```diff
@@ -0,0 +1,508 @@
+                                    sequences: SequenceDictionary,
+                                    sequenceDictionaryId: String): Unit = {
+
+    val conf = HBaseConfiguration.create()
+    val connection = ConnectionFactory.createConnection(conf)
+    val table: Table = connection.getTable(TableName.valueOf(hbaseTableName))
+    val contigs = sequences.toAvro
+
+    val baos: java.io.ByteArrayOutputStream = new ByteArrayOutputStream()
+    val encoder = EncoderFactory.get().binaryEncoder(baos, null)
+
+    contigs.foreach((x) => {
+      val contigDatumWriter: DatumWriter[Contig] =
+        new SpecificDatumWriter[Contig](scala.reflect.classTag[Contig].runtimeClass.asInstanceOf[Class[Contig]])
+      contigDatumWriter.write(x, encoder)
+    })
+
+    encoder.flush()
+    baos.flush()
+
+    val put: Put = new Put(Bytes.toBytes(sequenceDictionaryId))
+    put.addColumn(Bytes.toBytes("meta"), Bytes.toBytes("contigdata"), baos.toByteArray)
+    table.put(put)
+  }
+
+  def loadSequenceDictionaryFromHBase(hbaseTableName: String,
+                                      sequenceDictionaryId: String): SequenceDictionary = {
+
+    val conf = HBaseConfiguration.create()
+    val connection = ConnectionFactory.createConnection(conf)
+    val table = connection.getTable(TableName.valueOf(hbaseTableName))
+
+    val myGet = new Get(Bytes.toBytes(sequenceDictionaryId))
+    val result = table.get(myGet)
+    val dataBytes = result.getValue(Bytes.toBytes("meta"), Bytes.toBytes("contigdata"))
+
+    val contigDatumReader: DatumReader[Contig] =
+      new SpecificDatumReader[Contig](scala.reflect.classTag[Contig]
+        .runtimeClass
+        .asInstanceOf[Class[Contig]])
+
+    val decoder = DecoderFactory.get().binaryDecoder(dataBytes, null)
+    var resultList = new ListBuffer[Contig]
+    while (!decoder.isEnd) {
+      resultList += contigDatumReader.read(null, decoder)
+    }
+
+    SequenceDictionary.fromAvro(resultList)
+
+  }
+
+  def saveSampleMetadataToHBase(hbaseTableName: String,
+                                samples: Seq[Sample]): Unit = {
+
+    val conf = HBaseConfiguration.create()
+    val connection = ConnectionFactory.createConnection(conf)
+    val table = connection.getTable(TableName.valueOf(hbaseTableName))
+
+    samples.foreach((x) => {
+      val baos: java.io.ByteArrayOutputStream = new ByteArrayOutputStream()
+      val encoder = EncoderFactory.get().binaryEncoder(baos, null)
+
+      val sampleDatumWriter: DatumWriter[Sample] =
+        new SpecificDatumWriter[Sample](scala.reflect.classTag[Sample].runtimeClass.asInstanceOf[Class[Sample]])
+      sampleDatumWriter.write(x, encoder)
+      encoder.flush()
+      baos.flush()
+
+      val curr_sampleid = x.getSampleId
+
+      val put = new Put(Bytes.toBytes(curr_sampleid))
+      put.addColumn(Bytes.toBytes("meta"), Bytes.toBytes("sampledata"), baos.toByteArray)
+      table.put(put)
+
+    })
+  }
+
+  def loadSampleMetadataFromHBase(hbaseTableName: String,
```

Comment:
I kinda prefer the `loadSampleMetadataFromHBase` name, since to me, `loadSamplesFromHBase` sounds more like "load the genotypes for samples x,y,z from HBase", instead of "load the sample descriptions for samples x,y,z from HBase".